test/unit/link_checker_test.rb

```diff
@@ -0,0 +1,94 @@
+require "test_helper"
+require "gds_api/test_helpers/link_checker_api"
+
+class LinkCheckerTest < ActiveSupport::TestCase
+  include GdsApi::TestHelpers::LinkCheckerApi
+
+  def url
+    @url ||= "http://example.com"
+  end
+
+  def text
+    %(There is a link <a href="#{url}">here</a>)
+  end
+
+  def data
+    @data ||= text
+  end
+
+  def link_checker
+    @link_checker ||= LinkChecker.new(data)
+  end
+
+  def fixtures_path
+    Rails.root.join "test/fixtures/smart_answer_flows/flow-with-links"
+  end
+
+  context "#urls" do
+    should "return set of urls in text" do
+      assert_equal Set[url], link_checker.urls
+    end
+
+    context "when text in an array" do
+      should "return set of urls in text" do
+        @data = [text]
+        assert_equal Set[url], link_checker.urls
+      end
+    end
+
+    context "when text contains no links" do
+      should "return an empty set" do
+        @data = "No links"
+        assert_equal Set[], link_checker.urls
+      end
+    end
+
+    context "when data missing" do
+      should "return an empty set" do
+        @link_checker = LinkChecker.new(nil)
+        assert_equal Set[], link_checker.urls
+      end
+    end
+  end
```

Comment:
Can we test different edges cases of text where the URL might be in markdown, HTML a tag, or include brackets in the URL?

Reply:
Done